master/internal/job/job.go

```diff
@@ -210,38 +163,34 @@ func (j *Manager) UpdateJobQueue(updates []*jobv1.QueueControl) error {
 	errors := make([]string, 0)
 	for _, update := range updates {
 		jobID := model.JobID(update.JobId)
-		jobActor := j.jobRef(jobID)
-		if jobActor == nil {
+		s := j.serviceByID[jobID]
```

Comment:
looks like we will need to mutex lock this function. we should probably update job.go to lock over public functions but not the private ones

Reply:
changed, but out of curiosity why lock public functions over the private ones?